.github/actions/build/action.yml

```diff
@@ -26,4 +30,9 @@ runs:
         cmake \
           --build ${build_dir} \
           --config ${{ inputs.configuration }} \
-          --parallel ${NUM_PROCESSORS:-$(nproc)}
+          --parallel ${NUM_PROCESSORS:-$(nproc)} \
+          --target ${{ inputs.cmake-target }}
+        [[ -n "${{ inputs.artifact }}" ]] \
+          && [[ -e "${build_dir}/${{ inputs.artifact }}" ]] \
+          && mv "${build_dir}/${{ inputs.artifact }}" ./ \
```

Comment:
This condition has three terms. Why would the second term ever be false when the first term true?

Reply:
in case of action being misconfigured. You make a good point, though - if the action is misconfigured, it will be preferable to fail, rather than silently skip moving the artifact. I will change.

Reply:
I'm wondering: why not just add a separate step, in the jobs that use this option, to move the artifact? It seems like the coverage report job doesn't even need to move the report to upload it. Can we skip moving it altogether and remove that step from this action?

Reply:
will do that, simpler is better